### PR TITLE
Migrate to codecov.io

### PR DESCRIPTION
**Pull request by junczhuMSFT**
Fix: #233 

Codecov Migrate Steps:

- [x] add coverage percent at .codecov
- [x] add codecov-action at build.yml
- [x] add badge at README.md

- [x] update make test with go test at makefile
- [x] update .gitignore / rmv .cover/ .test/ add coverage.txt

- [x] del shell/test.sh
- [x] del origin upload coverage report from build.yml

Signed-off-by: Juncheng Zhu <junczhu@microsoft.com>

### CODE REVIEW

**Comment by junczhuMSFT** [comment]
Need access to get the badge for README.md
REF: https://docs.codecov.com/docs/status-badges

File: Makefile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 .PHONY: test
 test: vendor check-encoding
-	./scripts/test.sh
+		go test -race -v -coverprofile=coverage.txt -covermode=atomic ./...
```
Comment (shizhMSFT) [inline Makefile:3]: There's extra space in the code.

Comment (junczhuMSFT) [reply Makefile]: Extra space has been removed now

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**